gateway/protocol.go

```diff
@@ -50,6 +51,11 @@ type Protocol interface {
 	// 	- Returns the list of all service IDs for which the gateway is configured to serve.
 	ConfiguredServiceIDs() map[protocol.ServiceID]struct{}
 
+	// GetSanctionedEndpoints returns the sanctioned endpoints for a given service ID.
+	// It is called by the router to allow quick information about currently sanctioned endpoints.
+	// This will eventually be removed in favour of a metrics-based approach.
```

Comment:
IMO it won't be removed. It's a debugging tool.

If it will be removed, add a TODO_TECHDEBT.

Reply:
Resolving all of these comments as I've majorly reworked this PR and they are all outdated.